components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
-				String[] blacklist = blackListWhiteListService.loadBlacklisted(searchDTO);
-				StringBuilder successMSG = new StringBuilder();
 
-				successMSG.append("{").append("\"Success\":").append("{").append("\"messageId\":\"")
-						.append("Blacklist result").append("\",");
-				successMSG.append("\"text\":\"").append("Blacklist numbers were retrieved from the system".intern())
-						.append("\",").append("\"variables\":");
-				successMSG.append(gson.toJson(blacklist)).append("}}");
+				BlacklistWhitelistCountResponseDTO responseDTO = blackListWhiteListService.count(dto);
+				StringBuilder successMSG = new StringBuilder();
 
-				return Response.status(Response.Status.OK).entity(successMSG.toString()).build();
+				return Response.status(Response.Status.OK).entity(gson.toJson(responseDTO)).build();
 			} catch (BusinessException e) {
-				return Response.status(Response.Status.BAD_REQUEST).entity(e.getErrorType()).build();
+				return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getErrorType()).build();
 			}
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(errorMSG.toString()).build();
-		}
+
 	}
 
-	/**
-	 * Removes from the blacklist
-	 *
-	 * { "apiName":"USSD" }
-	 */
-	@POST
-	@Path("/RemoveFromBlacklist/{MSISDN}")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response removeFromBlacklist(@PathParam("MSISDN") String msisdn, String jsonBody) throws SQLException {
-		LOG.debug("removeFromBlacklist Triggerd  jsonBody :" + jsonBody + " , msisdn :" + msisdn);
+	@GET
+	@Path("/downloadAsZip/{app}/{api}")
+	@Produces("application/zip")
+	public Response getBlacklistAsZip(@PathParam("app") String app, @PathParam("api") String api, @QueryParam("action") String action, @QueryParam("sp") String sp) {
 
-		Gson gson = new GsonBuilder().serializeNulls().create();
+		app = replaceAllWithWildcard(app);
+		api = replaceAllWithWildcard(api);
+		sp = replaceAllWithWildcard(sp);
 
-		RemoveRequest removeReq = gson.fromJson(jsonBody, RemoveRequest.class);
+		final BlacklistWhitelistDTO dto = new BlacklistWhitelistDTO();
+		dto.setAppId(app);
+		dto.setApiID(api);
+		dto.setAction(action);
+		dto.setServiceProvider(sp);
 
-		StringBuilder erroMsg = new StringBuilder();
-		erroMsg.append("{").append("\"Failed\":").append("{").append("\"messageId\":\"").append("Remove Number")
-				.append("\",").append("\"text\":\"");
-		erroMsg.append("Blacklist number could not be removed ".intern()).append("\",").append("\"variables\":\"")
-				.append(msisdn).append("\"").append("}}");
+		try {
 
-		int apiId = removeReq.getAPIId();
+				StreamingOutput stream = new StreamingOutput() {
+					@Override
+					public void write(OutputStream outputStream) throws IOException, WebApplicationException {
 
-		if (apiId != -1) {
+						try {
+							blackListWhiteListService.getBlacklistAsZip(dto, outputStream);
+						} catch (Exception e) {
+							throw new WebApplicationException(e);
```

Comment:
log the exception here

Reply:
exception is logged by outer catch block